pkg/presentation/controllers/login_controller.go

```diff
@@ -30,31 +70,61 @@ func (c *LoginController) Register(r *mux.Router) {
 }
 
 func (c *LoginController) Get(w http.ResponseWriter, r *http.Request) {
+	email := r.URL.Query().Get("email")
+	errorsMap, err := composables.UseFlashMap[string, string](w, r, "errorsMap")
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	errorMessage, err := composables.UseFlash(w, r, "error")
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
 	pageCtx, err := composables.UsePageCtx(r, types.NewPageData("Login.Meta.Title", ""))
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	if err := login.Index(pageCtx).Render(r.Context(), w); err != nil {
+	if err := login.Index(&login.LoginProps{
+		PageContext:  pageCtx,
+		ErrorsMap:    errorsMap,
+		Email:        email,
+		ErrorMessage: string(errorMessage),
+	}).Render(r.Context(), w); err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
 
 func (c *LoginController) Post(w http.ResponseWriter, r *http.Request) {
-	if err := r.ParseForm(); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	dto := LoginDTO{
```

Comment:
It's be better if you switched to using Decoder/Validator as we do in other contorollers.